librecad/src/lib/engine/rs_hatch.cpp

```diff
@@ -607,7 +607,7 @@ void RS_Hatch::draw(RS_Painter* painter, RS_GraphicView* view, double& /*pattern
 //                    if (! (pa.size()>0 && (pa.last() - pt1).manhattanLength()<=2)) {
 //                        jp<<pt1;
 //                    }
-                    if(pa.size() && (pa.last()-pt1).manhattanLength()>=1)
+                    //if(pa.size() && (pa.last()-pt1).manhattanLength()>=1)
```

Comment:
This condition should avoid to store points twice. A container containing a bunch of lines have end points matching with the start point of the following entity. Storing all points will double the size of the resulting polygon.
So the condition has its right to exist, but it failed on the very first start point which led to the issue.